Fixed #35487 -- Removed CASCADE from RemoveField() on PostgreSQL. (#18209)
django/django: docs/ref/migration-operations.txt

populate the recreated column. If the field is not nullable and does not have a
default value, the operation is irreversible.

> .. admonition:: PostgreSQL

Reviewer: > I've added test but not sure about "cleanliness" as I've discovered that mysql is not restricting column drop used by view 😬. 

I wonder if we could adapt this admonition for the MySQL case as it appears that the following is True
```
``RemoveField`` will also delete any additional database objects that are
related to the removed field (like views, for example).
```

    :class:`~django.db.backends.base.schema.BaseDatabaseSchemaEditor` and
    PostgreSQL backends no longer use ``CASCADE`` to delete dependent related
    database objects, such as views. Any dependent objects that are not managed

>     by Django have to be removed another way before ``RemoveField`` runs.

Reviewer: Unsure if we should mention the MySQL exception

>     by Django have to be removed another way before ``RemoveField`` runs.

Reviewer 2: This may be an issue on other database not only on MySQL. I'd change wording to cover all cases without listing a concrete vendors, e.g.
```suggestion
    by Django may need be removed manually before running ``RemoveField``.
```

populate the recreated column. If the field is not nullable and does not have a
default value, the operation is irreversible.

> .. admonition:: PostgreSQL

Reviewer 3: I think we want a `.. versionchanged:: 5.2` block here to explain that `CASCADE` is no longer used in the base backend and PostgreSQL.

.. _dependent objects outside the table are also dropped: https://www.postgresql.org/docs/current/sql-altertable.html#SQL-ALTERTABLE-PARMS-CASCADE

> .. versionchanged:: 5.2

Reviewer 4: We should remove the mention just above now I believe?

Reviewer 3: Yeah, it was removed before but looks like it accidentally came back when adding the `versionchanged` annotation.

( https://github.com/django/django/pull/18209#discussion_r1618714748 )

Author: Yes, sorry for that, my bad. Fixed in latest commit.

>     The base and PostgreSQL database backends no longer use ``CASCADE`` to
>     avoid unintentionally side effects. Dependent objects unmanaged by
>     Django itself have to be removed manually before ``RemoveField``.

Reviewer 3: Attempt to make it clearer:

```suggestion
    The base and PostgreSQL database backends no longer use ``CASCADE`` to
    delete dependent related database objects, such as views. Any dependent
    objects that are not managed by Django have to be removed another way
    before ``RemoveField`` runs.
```
